src/main/kotlin/com/atlassian/performance/tools/infrastructure/Pidstat.kt

```diff
@@ -0,0 +1,35 @@
+package com.atlassian.performance.tools.infrastructure
+
+import com.atlassian.performance.tools.infrastructure.api.os.MonitoringProcess
+import com.atlassian.performance.tools.infrastructure.api.os.OsMetric
+import com.atlassian.performance.tools.infrastructure.api.process.RemoteMonitoringProcess
+import com.atlassian.performance.tools.ssh.api.SshConnection
+
+internal class Pidstat private constructor(
+    private val logPath: String,
+    private val pidstatParams: String
+) : OsMetric {
+    override fun startMonitoring(
+        connection: SshConnection
+    ): MonitoringProcess {
+        throw NotImplementedError("Use `start` method.")
+    }
+
+    override fun start(ssh: SshConnection): RemoteMonitoringProcess {
+        val process = ssh.startProcess("pidstat $pidstatParams > $logPath 2>&1 > $logPath")
+        return MonitoringProcess(process, logPath)
+    }
+
+    class Builder {
+        private var logPath: String = "~/jpt-pidstat.log"
+        private var pidstatParams: String = "-drsuw -t 2"
+
+        fun logPath(logPath: String): Builder = apply { this.logPath = logPath }
+        fun params(pidstatParams: String): Builder = apply { this.pidstatParams = pidstatParams }
```

Comment:
Consistency + context:
```suggestion
        fun params(params: String): Builder = apply { this.params = params }
```